lib/algo/memoization/0-1 Knapsack/index.js

```diff
@@ -0,0 +1,27 @@
+function createMatrix(x, y) {
+  const matrix = [];
+  for (let i = 0; i < x; i++) {
+    matrix[i] = [...new Array(y)].map(() => -1);
+  }
+  return matrix;
+}
+
+function knapSack(items, currentIndex, residualCapacity, valuesMatrix = createMatrix(currentIndex + 1, residualCapacity + 1)) {
+  if (residualCapacity < 0) {
+    return Number.MIN_SAFE_INTEGER;
+  }
+  if (currentIndex < 0) {
+    return 0;
+  }
+  let currentItem = valuesMatrix[currentIndex][residualCapacity] || 0;
+  if (currentItem !== -1) {
+    return currentItem;
+  }
+  const taking = (knapSack(items, currentIndex - 1, residualCapacity - items[currentIndex].volume, valuesMatrix) + items[currentIndex].profit) || 0;
+  const notTaking = knapSack(items, currentIndex - 1, residualCapacity, valuesMatrix) || 0;
```

Comment:
You might not want to use `||` for default value, you can read [more at ](https://sarbbottam.github.io/javascript/2016/05/27/never-use-or-for-default-values/)